.github/workflows/updated_examples.yaml

```
@@ -8,9 +8,17 @@ jobs:
   generate-example-files:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout
+      - name: Debug PR info
+        run: |
+          echo "PR head ref: ${{ github.event.pull_request.head.ref }}"
+          echo "PR head repo: ${{ github.event.pull_request.head.repo.full_name }}"
+
+      - name: Checkout PR head branch
         uses: actions/checkout@v4
         with:
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.event.pull_request.head.ref }}
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Is the token is needed to check out forks of the repo?

If so, how about adding a quick note there so we remember why it's necessary, and as an extra precaution, adding:
```suggestion
          token: ${{ secrets.GITHUB_TOKEN }}
          persist-credentials: false
```
From what I've read, this prevents the token from being stored for 'future use' by any subsequent Git operations which keeps things more secure when checking out potentially untrusted code from forks.

https://github.com/actions/checkout?tab=readme-ov-file#checkout-v4